readme.md

```diff
@@ -140,63 +140,17 @@ We want to use `zos-lib` to deploy this contract with upgradeability capabilitie
-
-  // For each version, a directory keeps track of the project's contract implementations.
-  console.log(`Deploying application directory for version ${initialVersion}...`);
-  this.directory = await AppDirectory.new(0, txParams);
-  console.log(`Deployed application directory for initial version at ${this.directory.address}`);
-
-  // Initialize the package with the first contract directory.
-  console.log(`Adding version to package...`);
-  await this.package.addVersion(initialVersion, this.directory.address, txParams);
-  console.log(`Added application directory to package`);
-
-  // With a proxy factory and a package, the project's app manager is bootstrapped and ready for use.
-  console.log(`Deploying application manager...`);
-  this.appManager = await AppManager.new(this.package.address, initialVersion, this.factory.address, txParams);
-  console.log(`Deployed application manager at ${this.appManager.address}`);
-
+  log("<< Setting up AppManager >>")
+  const initialVersion = '0.0.1'
+  const appManager = AppManagerDeployer.call(initialVersion, txParams)
 ```
 
 Next, we need to deploy the first version of the app contracts. To do so, we register the implementation of our `DonationsV1` in the `AppManager` and request it to create a new upgradeable proxy for it. Let's do it:
 ```js
-  console.log("\n<< Deploying version 1 >>");
-
-  // Deploy an implementation that defines the behavior of the main contract.
-  console.log(`Deploying first implementation of ${contractName}...`);
-  const implementation = await DonationsV1.new(txParams);
-  console.log(`Deployed first implementation at ${implementation.address}`);
-
-  // Register the implementation in the current version of the app.
-  console.log(`Registering implementation...`);
-  await this.directory.setImplementation(contractName, implementation.address, txParams);
-  console.log(`Registered implementation in current contract directory`);
-
-  // Create a proxy that wraps the implementation, making it upgradeable.
-  // At this point, the proxy's address is usable by any dapp, but can also be upgraded
-  // without having to use a new address or losing the contract's storage.
-  console.log(`Creating proxy for ${contractName}...`);
-  const callData = encodeCall('initialize', ['address'], [owner]);
-  const {receipt} = await this.appManager.createAndCall(contractName, callData, txParams);
-  const logs = decodeLogs([receipt.logs[1]], UpgradeabilityProxyFactory, 0x0);
-  const proxyAddress = logs.find(l => l.event === 'ProxyCreated').args.proxy;
-  this.proxy = OwnedUpgradeabilityProxy.at(proxyAddress);
-  console.log(`Proxy for ${contractName} created at ${proxyAddress}`);
+  log("\n<< Deploying version 1 >>")
+  const DonationsV1 = ContractsProvider.getByName('DonationsV1')
+  await appManager.setImplementation(DonationsV1, contractName);
+  await appManager.createProxy(DonationsV1, contractName, 'initialize', [owner])
```

Comment:
Could you comment the steps so it's 100% clear for a beginner?

Reply:
Done. Please note that the comments are concise, but that they go accompanied by a tutorial in the readme.